src/com/googlecode/jmxtrans/model/output/GangliaWriter.java

```diff
@@ -166,11 +169,14 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 			if (result.getValues() != null) {
 				for (final Map.Entry<String, Object> resultValue : result.getValues().entrySet()) {
 					final String name = KeyUtils.getKeyString(query, result, resultValue, getTypeNames());
-					final String value = resultValue.getValue().toString();
+					
+					BigDecimal value = new BigDecimal(resultValue.getValue().toString());
```

Comment:
As you are adding a branch in the execution flow, it make sense to add a test for it as well.